src/App.tsx

```diff
@@ -29,128 +70,101 @@ export const App: React.FC = () => {
-              Active
-            </a>
-
-            <a href="#/completed" className="filter__link">
-              Completed
-            </a>
-          </nav>
-
-          {/* don't show this button if there are no completed todos */}
-          <button type="button" className="todoapp__clear-completed">
-            Clear completed
-          </button>
-        </footer>
+        {allTodos.length > 0 && (
+          <footer className="todoapp__footer">
+            <span className="todo-count">
+              {`${activeTodos.length} items left`}
+            </span>
+
+            <nav className="filter">
+              <a
+                href="#/"
+                className={classNames(
+                  'filter__link', { selected: filter === Filter.All },
+                )}
+                onClick={() => setFilter(Filter.All)}
+              >
+                All
+              </a>
+
+              <a
+                href="#/active"
+                className={classNames(
+                  'filter__link', { selected: filter === Filter.Active },
+                )}
+                onClick={() => setFilter(Filter.Active)}
+              >
+                Active
+              </a>
+
+              <a
+                href="#/completed"
+                className={classNames(
+                  'filter__link', { selected: filter === Filter.Completed },
+                )}
+                onClick={() => setFilter(Filter.Completed)}
+              >
+                Completed
+              </a>
+            </nav>
+
+            <button
+              type="button"
+              className="todoapp__clear-completed"
+              style={{ opacity: completedTodos.length === 0 ? 0 : 1 }}
+            >
+              Clear completed
+            </button>
+          </footer>
+        )}
       </div>
 
-      {/* Notification is shown in case of any error */}
-      {/* Add the 'hidden' class to hide the message smoothly */}
-      <div className="notification is-danger is-light has-text-weight-normal">
-        <button type="button" className="delete" />
-
-        {/* show only one message at a time */}
-        Unable to add a todo
-        <br />
-        Unable to delete a todo
-        <br />
-        Unable to update a todo
-      </div>
+      {isError && (
+        <div className={classNames(
+          'notification is-danger is-light has-text-weight-normal',
+          { hidden: isErrorHidden },
+        )}
+        >
+          <button
+            type="button"
+            className="delete"
+            onClick={() => setIsErrorHidden(true)}
+          />
+
+          {isError === Error.Load && 'Unable to load todos'}
+          <br />
+          {isError === Error.Add && 'Unable to add a todo'}
+          <br />
+          {isError === Error.Delete && 'Unable to delete a todo'}
+          <br />
+          {isError === Error.Update && 'Unable to update a todo'}
+        </div>
+      )}
```

Comment:
1. Move error block into component
2.  Create object with errors to make your code more comact
```suggestion
const Errors = {
  [Error.Load]: 'Unable to load todos',
  ....
};

......

      {isError && (
        <div className={classNames(
          'notification is-danger is-light has-text-weight-normal',
          { hidden: isErrorHidden },
        )}
        >
          <button
            type="button"
            className="delete"
            onClick={() => setIsErrorHidden(true)}
          />

          {Errors[isError]}
        </div>
      )}
```